src/lib.rs

```diff
@@ -115,6 +115,357 @@ impl WaveletTreeNode {
     }
 }
 
+pub struct PointerlessWaveletTree<T: PartialOrd + Clone> {
+    alphabet: Vec<T>,
+    bitmap: RankSelect
+}
+
+impl <T: PartialOrd + Clone> PointerlessWaveletTree<T> {
+
+    pub fn from_sequence(sequence: &[T]) -> Self {
+        // Create a vector for storing the alphabet of the sequence
+        let mut alphabet = Vec::new();
+
+        // Add all symbols from the sequence to the new alphabet vector
+        for symbol in sequence.iter() {
+            if !alphabet.contains(symbol) {
+                alphabet.push(symbol.clone());
+            }
+        }
+
+        // Sort the alphabet
+        alphabet.sort_by(|x,y| x.partial_cmp(y).unwrap());
+
+        let bitmap = Self::create_bits(sequence, &alphabet);
+
+        // Return a new PointerWaveletTree containing the alphabet and the root of the tree
+        PointerlessWaveletTree {
+            alphabet,
+            bitmap
+        }
+    }
+
+    /// Create new Pointerless Wavelet Tree
+    pub fn create_bits(sequence: &[T], alphabet: &[T]) -> RankSelect {
+        if sequence.is_empty() {
+            return RankSelect::new(BitVec::new(), 1);
+        }
+        else {
+            let mut bits: BitVec<u8> = BitVec::new();
+
+            // Special Case if Alphabet contains one Element
+            if alphabet.len() == 1 {
+                for i in 0..sequence.len() {bits.push(false);}
```

Comment:
Variable `i` is unused here. In my opinion, the following code would be a little bit more elegant:
```suggestion
                bits = BitVec::new_fill(false, sequence.len());
```